playwright/README.md

```diff
@@ -0,0 +1,75 @@
+
+```mermaid
+graph TD
+    A[Playwright] --> W1
+    A --> W2
+    A --> WN
+    subgraph Workers
+      W1[Worker1]
+      W2[Worker2]
+      WN[WorkerN]
+    end
+    W1[Worker1] --> C
+    W2[Worker2] -->|http://localhost:4000| C
+    WN[WorkerN] --> C
+    C[Web dev server] -->|http://localhost:10000| D[deephaven-core server]
+```
+
+## Running Tests
+
+Playwright tests can be run by executing `npx playwright test`. By default they are executed in Headless mode, meaning the browser is not visible or open.
+
+## Viewing Report
+
+By default, after running the tests, if there is a failure you will be able to view the report:
+
+![Report console](./assets/report-console.png)
+
+Opening the report will give you details on which tests are failing:
+
+![Report view](./assets/report-view.png)
+
+You can then open the details of each report, showing which step within your test failed, along with a video to help with debugging:
+
+![Report details](./assets/report-details.png)
+
+## Writing Tests
+
+We use React Testing Library to write our unit tests. Playwright uses slightly different syntax, but there is a [Cheat Sheet](https://playwright.dev/docs/testing-library#cheat-sheet) that details how to write equivalent tests in Playwright which can be handy.
+
+### Codegen
+
+You can use a generator to write the tests for you. Run `npx playwright codegen http://localhost:4000`, and then perform the actions you would like to do in your tests, and code will be generated for you. After generating tests, you may want to clean up some of the selectors, and/or write some `expect` checks to verify what is happening on the screen.
+
+![Codegen](./assets/codegen.gif)
+
+Often the selectors generated from codegen are somewhat convoluted, so you will want to review them to see if they can be handled differently.
+
+### Screenshots
+
+Some elements (such as our Grid or plots) are canvas based elements, and we cannot verify the contents of the elements through the DOM alone. In these cases, we can use [test snapshots](https://playwright.dev/docs/test-snapshots). An example of checking a grid element for a screenshot:
+
+```
+  await expect(page.locator('.iris-grid-panel .iris-grid')).toHaveScreenshot();
+```
+
+By default, Playwright will generate the screenshots on first run. If you need to force screenshots to update, you can run `npx playwright test --update-snapshots`.
+
+TODO: For CI, we also need to generate screenshots matching the OS that will be used in CI. Because CI may run in a different OS/Environment, we can [create a docker container to generate the snapshots](https://playwright.dev/docs/test-snapshots).
```

Comment:
Is the word "TODO:" intentional here?

Reply:
Removed the word TODO, we'll need to investigate how to set up the Docker container.